### DISCUSSION TITLE
How to connect using reactive drivers

### DISCUSSION THREAD

**Discussion by MiguelAngelOne** [planetscale/discussion#553] [Q&A] [answered]
Hi I'm using quakus for my project and I want to connect to the database via reactive drivers, but it always gives me this error
```
{
  "details": "Error id eeeb6a93-6c18-40d3-90ea-c308ee980dd0-1, io.vertx.mysqlclient.MySQLException: {errorMessage=unknown error: Code: UNAVAILABLE",
  "stack": "io.vertx.mysqlclient.MySQLException: {errorMessage=unknown error: Code: UNAVAILABLE\nserver does not allow insecure connections, client must use SSL/TLS\n, errorCode=1105, sqlState=HY000}"
}
```

I have my property vars like this
```
# datasource
quarkus.datasource.db-kind=mysql
quarkus.datasource.username=QUARKUS_DATASOURCE_USERNAME
quarkus.datasource.password=QUARKUS_DATASOURCE_PASSWORD
quarkus.datasource.reactive.url=vertx-reactive:mysql://planetscale-host/db-name?sslMode=VERIFY_IDENTITY

```

I tested with jdbc drivers an is working fine, but when I use reactive it does not work, any help would be appreciated.

Thanks

**Comment by jgreet** [comment]
Interesting, I don't think we've looked at Quarkus before.

These docs suggest you can set the SSL mode with the property `quarkus.datasource.reactive.mysql.ssl-mode`. Can you try setting `verify-identity` there instead?

https://quarkus.io/guides/datasource#reactive-mariadbmysql-specific-configuration

**Comment by MiguelAngelOne** [comment, marked as answer]
Hi, jgreet,

I forgot to update, I found the correct configuration, here it is what is working for me, you also have to specify the verification algoritm.

```
quarkus.datasource.db-kind=mysql
quarkus.datasource.reactive.mysql.ssl-mode=VERIFY_IDENTITY
quarkus.datasource.reactive.hostname-verification-algorithm=HTTPS
quarkus.datasource.username=QUARKUS_DATASOURCE_USERNAME
quarkus.datasource.password=QUARKUS_DATASOURCE_PASSWORD
quarkus.datasource.reactive.url=vertx-reactive:mysql://planetscale-host/db-name
```

**Reply by jgreet** [reply]
Thanks for sharing!
